joblib/test/test_hashing.py

```diff
@@ -320,3 +322,18 @@ def test_set_hash():
     b = k.f(a)
 
     nose.tools.assert_equal(hash(a), hash(b))
+
+
+def test_string():
```

Comment:
Please explain, in a comment, what these tests actually test (the thing about the identity isn't trivial).